sdks/python/apache_beam/runners/pipeline_context.py

```diff
@@ -158,13 +150,26 @@ def __init__(self,
           coders=dict(proto.coders.items()),
           windowing_strategies=dict(proto.windowing_strategies.items()),
           environments=dict(proto.environments.items()))
-    for name, cls in self._COMPONENT_TYPES.items():
-      setattr(
-          self, name, _PipelineContextMap(
-              self, cls, namespace, getattr(proto, name, None)))
+
+    self.transforms = _PipelineContextMap(
```

Comment:
This feels very DRY...

Reply:
This is kind of the nature of static type checking:  you can't do dynamic attribute definition if you want to accurately track the types of those attributes.   I'm open to other suggestions.